sdk-cpp/src/internal/com/do_download_property_internal.cpp

```
@@ -19,77 +19,85 @@ std::wstring UTF8toWstr(const char* str, size_t cch = 0)
     }
     V_BSTR(&_var) = bstr;
+
+    return S_OK;
 };
 
-CDownloadPropertyValueInternal::CDownloadPropertyValueInternal(uint32_t val)
+int32_t CDownloadPropertyValueInternal::Init(uint32_t val) noexcept
 {
     V_VT(&_var) = VT_UI4;
     V_UI4(&_var) = val;
+    return S_OK;
 };
 
-CDownloadPropertyValueInternal::CDownloadPropertyValueInternal(uint64_t val)
+int32_t CDownloadPropertyValueInternal::Init(uint64_t val) noexcept
 {
     V_VT(&_var) = VT_UI8;
     V_UI8(&_var) = val;
+    return S_OK;
 };
 
-CDownloadPropertyValueInternal::CDownloadPropertyValueInternal(bool val)
+int32_t CDownloadPropertyValueInternal::Init(bool val) noexcept
 {
     V_VT(&_var) = VT_BOOL;
     V_BOOL(&_var) = val ? VARIANT_TRUE : VARIANT_FALSE;
+    return S_OK;
 };
 
-CDownloadPropertyValueInternal::CDownloadPropertyValueInternal(std::vector<unsigned char>& val)
+int32_t CDownloadPropertyValueInternal::Init(std::vector<unsigned char>& val) noexcept
 {
-    throw errc::e_not_impl;
+    return static_cast<int32_t>(errc::e_not_impl);
 };
 
-CDownloadPropertyValueInternal::CDownloadPropertyValueInternal(const download_property_value::status_callback_t& val)
+int32_t CDownloadPropertyValueInternal::Init(const download_property_value::status_callback_t& val) noexcept
 {
+    V_VT(&_var) = VT_EMPTY;
     _callback = val;
+    return S_OK;
 }
 
 CDownloadPropertyValueInternal::~CDownloadPropertyValueInternal()
 {
 #ifdef DEBUG
-    if (!_callback)
-    {
-        assert(SUCCEEDED(VariantClear(&_var)));
-    }
+    assert(SUCCEEDED(VariantClear(&_var)));
 #else
     (void)VariantClear(&_var);
 #endif
 };
 
 CDownloadPropertyValueInternal::CDownloadPropertyValueInternal(const CDownloadPropertyValueInternal& rhs)
 {
-    const auto res = VariantCopy(&_var, &rhs._var);
-#if (!DO_DISABLE_EXCEPTIONS)
-    microsoft::deliveryoptimization::throw_if_fail(res);
+#ifdef DEBUG
+    assert(SUCCEEDED(VariantCopy(&_var, &rhs._var)));
+#else:
+    (void)VariantCopy(&_var, &rhs._var);
```

>void [](http://example.com/codeflow?start=5&length=4)

Silently ignores errors. I suggest calling std::terminate instead when this fails (memory alloc failure).

---

Noted, fixing in next PR